includes/class-newspack-ads-scaip.php

```diff
@@ -0,0 +1,97 @@
+<?php
+/**
+ * Newspack Ads SCAIP Hooks
+ *
+ * @package Newspack
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Newspack Ads SCAIP Class.
+ */
+class Newspack_Ads_SCAIP {
+
+	// Map of SCAIP option names.
+	const OPTIONS_MAP = array(
+		'start'          => 'scaip_settings_start',
+		'period'         => 'scaip_settings_period',
+		'repetitions'    => 'scaip_settings_repetitions',
+		'min_paragraphs' => 'scaip_settings_min_paragraphs',
+	);
+
+	/**
+	 * Initialize SCAIP Hooks.
+	 */
+	public static function init() {
+		add_filter( 'newspack_ads_settings_list', array( __CLASS__, 'add_settings' ) );
+		add_filter( 'newspack_ads_setting_option_name', array( __CLASS__, 'map_option_name' ), 10, 2 );
+	}
+
+	/**
+	 * Add SCAIP settings to the list of settings.
+	 *
+	 * @param array $settings_list List of settings.
+	 *
+	 * @return array Updated list of settings.
+	 */
+	public static function add_settings( $settings_list ) {
+
+		if ( ! defined( 'SCAIP_PLUGIN_FILE' ) ) {
+			return $settings_list;
+		}
+
+		$scaip_settings = array(
+			array(
+				'description' => __( 'Post ad inserter settings', 'newspack-ads' ),
+				'help'        => __( 'Super Cool Ad Inserter plugin options', 'newspack-ads' ),
+				'section'     => 'scaip',
+			),
+			array(
+				'description' => __( 'Number of blocks before first insertion', 'newspack-ads' ),
```

Comment:
As now ads are inserted on count of paragraphs, better to change wordings to paragraph, instead of blocks.

Reply:
This is a good point. Not sure if we should change it to "paragraph", though. The types of blocks are filterable, although they are only paragraphs by default.

If SCAIP upgrades to support the selection of allowed insertion blocks on the UI the "block" wording would make better sense.

cc @adekbadek